### ISSUE TITLE
Update to latest version of LambdaHack

### ISSUE THREAD

**Issue by tuturto** [tuturto/space-privateers#9]

**Comment by Mikolaj** [comment]
How is it going? When recently updating Allure of the Stars wrt LH changes, I've just started over from LH content, added Allure-specific content at the end of files and then reviewed and removed all the LH content that I didn't want or modified it (but not the Haskell names of the content, so that I could keep diffs small). If you post a branch with current progress, I bet I could drop you some hints.

**Comment by tuturto** [comment]
I haven't yet started, so it's not progressing :) Trying to set aside some time this weekend, so I can at least start, if not finish this during the weekend. Thanks for the tip.

**Comment by tuturto** [comment]
Having some issues with dependencies (commit cf5b850):

```
tuukka@azrael:~/programming/space-privateers$ cabal install
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: SpacePrivateers-0.1.3.0 (user goal)
trying: text-1.2.2.2 (dependency of SpacePrivateers-0.1.3.0)
trying: binary-0.7.1.0/installed-f86... (dependency of text-1.2.2.2)
next goal: LambdaHack (dependency of SpacePrivateers-0.1.3.0)
rejecting: LambdaHack-0.6.0.0 (conflict: binary==0.7.1.0/installed-f86...,
LambdaHack => binary>=0.8)
rejecting: LambdaHack-0.5.0.0, 0.4.101.1, 0.4.101.0, 0.4.100.0, 0.4.99.0,
0.4.9.0, 0.2.14, 0.2.12, 0.2.10.6, 0.2.10.5, 0.2.10, 0.2.8, 0.2.6.5, 0.2.6,
0.2.1, 0.2.0, 0.1.20110918, 0.1.20110117, 0.1.20090606, 0.1.20080413,
0.1.20080412 (conflict: SpacePrivateers => LambdaHack>=0.6.0.0 && <0.6.1.0)
Dependency tree exhaustively searched.

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox.
tuukka@azrael:~/programming/space-privateers$
```
Probably have to set upper bounds too.

**Comment by Mikolaj** [comment]
It says `conflict: binary==0.7.1.0/installed-f86...,LambdaHack => binary>=0.8`, which means you have binary 0.7.1.0 in your sandbox. I guess, the easiest course of action is to wipe out the sandbox (not the repo) and create a new one.

**Comment by tuturto** [comment]
Hm.. it's not even in the sandbox, but in opt directory:

```
tuukka@azrael:~/programming/space-privateers$ ghc-pkg list
/opt/ghc/7.8.4/lib/ghc-7.8.4/package.conf.d
...
   binary-0.7.1.0
...
```

I wonder why, since I should be using sandbox created with ```cabal sandbox init```.

```ghc --version``` reports following: ```The Glorious Glasgow Haskell Compilation System, version 7.8.4```. I'm suspecting that might be the problem.

**Comment by Mikolaj** [comment]
Oh, I see. I rarely use sandboxes and I have zero globally installed packages ever, I just boostrap from GHC and cabal executable in my home directory, so I'm not sure what to do in this case. Perhaps manually install the new version of `binary`?

**Comment by tuturto** [comment]
I'll look into bootstrapping in same manner as you do later in the evening as the ghc that I have installed is fairly old.

**Comment by Mikolaj** [comment]
True. Just download and install https://www.haskell.org/ghc/download_ghc_8_0_2.html, copy over the cabal exe from your old location to somewhere where you can execute it, wipe all old stuff out, including ~/.ghc and ~/.cabal, do 'cabal install cabal-install' and you are set. :)

**Comment by tuturto** [comment]
That was the problem. I installed latest GHC and could then ```cabal install --dependencies-only``` without problems. Now I can continue with the original task, updating to the latest LambdaHack version.
